include/aspect/material_model/visco_plastic.h

```diff
@@ -168,11 +168,36 @@ namespace aspect
 
         double grain_size;
 
-        std::vector<double> densities;
+        // list of depth (or pressure), width and Clapeyron slopes
+        // for the different phase transitions
+        std::vector<double> transition_depths;
+        std::vector<double> transition_pressures;
+        std::vector<double> transition_temperatures;
+        std::vector<double> transition_widths;
+        std::vector<double> transition_pressure_widths;
+        std::vector<double> transition_slopes;
+
+        // Number of phase changes
+        unsigned int number_of_phase_transitions;
+
+        // Use depth (if true) instead of pressure to define phase transitions
+        bool use_depth;
+
+        dealii::Table<2,double> densities;
         std::vector<double> thermal_expansivities;
         std::vector<double> thermal_diffusivities;
         std::vector<double> heat_capacities;
 
+        /**
+         * Percentage of material that has already undergone the phase
+         * transition to the higher-pressure material (this is done
+         * individually for each transition and summed up in the end)
+         */
```

Comment:
This is the documentation of the original function in the latent heat model, but now you don't just return a number representing a percentage of material that has already undergone the phase transition, you return a vector that represents the percentage of each phase that is present at a given temperature and pressure. This is a different thing, and is also why the computation doesn't work at the moment if more than 2 phases are present.

Reply:
Looking back, I don't recall exactly why I structured the phase function calculation differently than in the latent heat model. I'll try rewriting this in the same manner as the latent  heat model and see if any issues pop up. If not, no reason to use the structure in this PR.

Reply:
Now the original documentation should be correct.